File: array/1365_how_many_numbers_are_smaller_than_the_current_number/1365_how_many_numbers_are_smaller_than_the_current_number.go
package main

func smallerNumbersThanCurrent(nums []int) []int {
	var sli = []int{}
	count := 0
	length := len(nums)
	for i := 0; i < length; i++ {
		for j := 0; j < length; j++ {
			if i == j {
				continue
			} else if nums[i] > nums[j] {
				count++
			}
		}
		sli = append(sli, count)
		count = 0
	}
	return sli
}

File: array/1365_how_many_numbers_are_smaller_than_the_current_number/1365_how_many_numbers_are_smaller_than_the_current_number.py

class Solution(object):
    def smallerNumbersTHanCurrent(self, nums):
        nums2 = sorted(nums)
        mapping = {}
        for i, num in enumerate(nums2):
            if i > 0 and nums2[i] == nums2[i-1]:
                mapping[nums2[i]] = mapping[nums2[i-1]]
            else:
                mapping[nums2[i]] = i
        res = []
        for num in nums:
            res.append(mapping[num])
        return res